lib/src/json_schema/json_schema.dart

```diff
@@ -1692,6 +1708,9 @@ class JsonSchema {
 
         // If the $id has a fragment, append it to the base, or use it alone.
       } else if (_id.fragment != null && _id.fragment.isNotEmpty) {
+        if (schemaVersion.hashCode >= SchemaVersion.draft2019_09.hashCode) {
```

Comment:
Woah, why are we comparing hash codes like this?

Reply:
The hash code is just the version number. There should be a nicer way of doing it soon.

Reply:
What an... amazing hashcode we have! /s